foundation/device/device-client/README.md

```diff
@@ -0,0 +1,708 @@
+            deviceClient.pushDevice.delete(deviceToDelete).onSuccess {
+                println("Device deleted successfully!")
+            }.onFailure { exception ->
+                println("Failed to delete device: ${exception.message}")
+            }
+        }
+    }
+}
+```
+
+**Note:** All device types support the `delete()` method through the unified `DeviceRepository`.
+
+## Integration with Journey and DaVinci
+
+The Device Client module is designed to work seamlessly with the Journey and DaVinci modules.
+
+### Integration with Journey
+
+After successful authentication using Journey, retrieve the SSO token and use it to initialize the Device Client:
+
+```kotlin
+import com.pingidentity.journey.Journey
+import com.pingidentity.device.client.DeviceClient
+import java.net.URL
+
+val journey = Journey {
+    serverUrl = URL("https://openam.example.com/am")
+    realm = "alpha"
+}
+
+// Authenticate the user
+var node = journey.start("login")
+// ... Complete the authentication flow
+
+if (node is SuccessNode) {
+    val session = node.session
+    val ssoToken = session.value // Extract the SSO token
+    
+    // Create DeviceClient with the SSO token
+    val deviceClient = DeviceClient {
+        ssoTokenString = ssoToken
+        serverUrl = URL("https://openam.example.com/am")
+        realm = "root"
+        cookieName = "iPlanetDirectoryPro"
+    }
+    
+    // Retrieve devices - userId is automatically fetched and cached
+    deviceClient.oathDevice.devices().onSuccess { devices ->
+        // Display devices to the user
+        println("Found ${devices.size} OATH devices")
+    }.onFailure { exception ->
+        println("Error: ${exception.message}")
+    }
+}
+```
+
+### Integration with DaVinci
+
+Similarly, after authenticating with DaVinci, use the user's session to create the Device Client:
+
+```kotlin
+import com.pingidentity.davinci.DaVinci
+import com.pingidentity.davinci.module.Oidc
+import com.pingidentity.device.client.DeviceClient
+import java.net.URL
+
+val daVinci = DaVinci {
+    module(Oidc) {
+        clientId = "your_client_id"
+        discoveryEndpoint = "https://auth.pingone.com/.../.well-known/openid-configuration"
+        redirectUri = "com.example.app://callback"
+        scopes = mutableSetOf("openid", "profile", "email")
+    }
+}
+
+var node = daVinci.start()
+// ... Complete the authentication flow
+
+if (node is SuccessNode) {
+    val user = daVinci.user()
+    val accessToken = user?.accessToken()
+    
+    // Create DeviceClient with the access token
+    val deviceClient = DeviceClient {
+        ssoTokenString = accessToken?.value
+        serverUrl = URL("https://openam.example.com/am")
+        realm = "root"
+        cookieName = "iPlanetDirectoryPro"
+    }
+    
+    // Retrieve devices - userId is automatically fetched and cached
+    val devices = deviceClient.boundDevice.devices()
+    // Display devices to the user
+}
+```
+
+
+### Key Features of this Implementation
```

Comment:
This section seems to be relevant to our sample app. I think we should remove it from this readme...